pkg/storage/chunk/client/aws/s3_storage_client.go

```diff
@@ -367,25 +368,30 @@ func (a *S3ObjectClient) bucketFromKey(key string) string {
 
 // GetObject returns a reader and the size for the specified object key from the configured S3 bucket.
 func (a *S3ObjectClient) GetObject(ctx context.Context, objectKey string) (io.ReadCloser, int64, error) {
-	var resp *s3.GetObjectOutput
-
 	// Map the key into a bucket
 	bucket := a.bucketFromKey(objectKey)
 
-	retries := backoff.New(ctx, a.cfg.BackoffConfig)
 	err := ctx.Err()
```

Comment:
Do you still need to set this to ctx.Err()? It is not used until another value is set for this field.

Reply:
it is used at line 398, but not sure if it is really possible to have a ctx.Err without ever entering the `retries.Ongoing` block. Maybe if a deadline is exceeded in the moment this function is invoked? :thinking:

Reply:
In line 393 the error is set. It isn't checked before that. So there's no point to set it to something.